### PR TITLE
Update Idea events

### PR DESCRIPTION
**Pull request by Pinkdesu**
- Update events logic

### CODE REVIEW

File: website/frontend/src/services/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { GearApi, Hex } from '@gear-js/api';
+import { UnsubscribePromise } from '@polkadot/api/types';
+
+export const subscribeToProgramUpload = (api: GearApi, programId: string) => {
```
Comment (osipov-mit) [inline website/frontend/src/services/helpers.ts:4]: lets call it `waitForProgramInit` or smth like this

File: website/frontend/src/types/explorer.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export enum Sections {
 
 export enum Methods {
   TRANSFER = 'Transfer',
-  LOG = 'Log',
+  USER_MESSAGE_SENT = 'UserMessageSent',
   INIT_SUCCESS = 'InitSuccess',
   INIT_FAILURE = 'InitFailure',
   DISPATCH_MESSAGE_ENQUEUED = 'DispatchMessageEnqueued',
```
Comment (osipov-mit) [inline website/frontend/src/types/explorer.ts:63]: This events don't exist from no on

Comment (osipov-mit) [reply website/frontend/src/types/explorer.ts:63]: And `MessageDispatched` is called `MessagesDispatched` now

File: website/frontend/src/components/pages/Explorer/children/Events/init.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { Methods } from 'types/explorer';
 
 export const filterValues = {
   [Methods.TRANSFER]: false,
-  [Methods.LOG]: false,
+  [Methods.USER_MESSAGE_SENT]: false,
   [Methods.INIT_SUCCESS]: false,
   [Methods.INIT_FAILURE]: false,
   [Methods.DISPATCH_MESSAGE_ENQUEUED]: false,
```
Comment (nikitayutanov) [inline website/frontend/src/components/pages/Explorer/children/Events/init.ts:8]: those methods are gone

File: website/frontend/src/components/pages/Meta/Meta.tsx [hunk 4]
```diff
@@ -9,7 +10,7 @@ import { PageHeader } from 'components/blocks/PageHeader/PageHeader';
 import { useProgram } from 'hooks';
 
 type Params = {
-  programId: string;
+  programId: Hex;
```
Comment (nikitayutanov) [inline website/frontend/src/components/pages/Meta/Meta.tsx:13]: don't think that it's right, cuz programId clearly can be regular string and not hex

File: website/frontend/src/services/ApiService.ts [hunk 5]
```diff
@@ -134,15 +146,26 @@ export const UploadProgram = async (
...
     });
+
+    const programUploadStatus = await getProgramUploadStatus();
+
+    if (programUploadStatus === 'failed') {
+      alert.error(PROGRAM_ERRORS.PROGRAM_INIT_FAILED);
+      return;
+    }
+
+    alert.success('Program initialization successfully');
```
Comment (nikitayutanov) [inline website/frontend/src/services/ApiService.ts:157]:
```suggestion
    alert.success('Program initializated successfully');
```

**Review by osipov-mit — APPROVED** [review]

**Review by nikitayutanov — APPROVED** [review]

**Review by aderrod94 — APPROVED** [review]

**Pull request merged**